homeassistant/components/googlehome/sensor.py

```diff
@@ -0,0 +1,109 @@
+"""
+Support for Google Home alarm sensor.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.googlehome/
+"""
+import logging
+from datetime import timedelta
+
+from homeassistant.components.googlehome import (
+    CLIENT, DOMAIN as GOOGLEHOME_DOMAIN, NAME)
+from homeassistant.helpers.entity import Entity
+import homeassistant.util.dt as dt_util
+
+
+DEPENDENCIES = ['googlehome']
+
+SCAN_INTERVAL = timedelta(seconds=10)
+
+_LOGGER = logging.getLogger(__name__)
+
+ICON = 'mdi:alarm'
+
+SENSOR_TYPES = {
+    'timer': "Timer",
+    'alarm': "Alarm",
+}
+
+
+async def async_setup_platform(hass, config,
+                               async_add_entities, discovery_info=None):
+    """Set up the googlehome sensor platform."""
+    if discovery_info is None:
+        _LOGGER.warning(
+            "To use this you need to configure the 'googlehome' component")
+        return False
+
+    devices = []
+    for condition in SENSOR_TYPES:
+        device = GoogleHomeAlarm(hass, hass.data[CLIENT], condition,
+                                 discovery_info)
+        await device.async_init()
+        devices.append(device)
+
+    async_add_entities(devices, True)
+
+
+class GoogleHomeAlarm(Entity):
+    """Representation of a GoogleHomeAlarm."""
+
+    def __init__(self, hass, client, condition, config):
+        """Initialize the GoogleHomeAlarm sensor."""
+        self._hass = hass
```

Comment:
all entities have the self.hass object (you can delete _hass)

Reply:
@dgomes, if I remove ```self._hass = hass``` and try to access ```hass``` with ```self.hass``` I get ```AttributeError: 'NoneType' object has no attribute 'data'```

Reply:
self.hass gets assign by async_added_to_hass() do you really need it before ?

Reply:
You can rename the `async_init` method to `async_added_to_hass`. Right as that method is called, it will have access to `self.hass`. You also should not manually call it, it will be called automatically.